zap/src/main/java/org/zaproxy/zap/extension/ascan/FilterPanelVO.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascan;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import org.zaproxy.zap.extension.ascan.filters.GenericFilterUtility;
+import org.zaproxy.zap.extension.ascan.filters.ScanFilter;
+
+/**
+ * FilterPanelVO is populated by {@link FilterPanel} and then passed to the {@link ActiveScan} where
+ * a Node is Filtered based on these Selections.
+ *
+ * @author KSASAN preetkaran20@gmail.com
+ * @since TODO add version
+ */
+public class FilterPanelVO {
```

Comment:
hmm but my point was UI/FilterPanel should not know about the internal implementations ie say ScanFilter implementation changes should not impact FilterPanel and also it would be better for UI to operate on one entity so that on Runtime changing that would be easy too.

Reply:
Practically, what's the difference in changing this class or the panel if the filter implementation changes?

> and also it would be better for UI to operate on one entity so that on Runtime changing that would be easy too.

The panel does not use any data from this class, were those changes not committed? Still, I don't think it would be responsibility of this class to provide/create the filters.

Reply:
hmm.... correct agreed. i thought that i update VO on every selection which i am not doing. will do that changes.